templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -188,30 +187,46 @@ contents:
     EOF
     }
 
-    # add member cluster
+    # add member to cluster
     etcd_member_add() {
       echo "Updating etcd membership.."
       if [ -d "$ETCD_DATA_DIR" ]; then
         echo "Removing etcd data_dir $ETCD_DATA_DIR.."
         rm -rf $ETCD_DATA_DIR
       fi
 
-      RESPONSE=$(env ETCDCTL_API=3 $ETCDCTL --cert $ASSET_DIR/backup/etcd-client.crt --key $ASSET_DIR/backup/etcd-client.key --cacert $ASSET_DIR/backup/etcd-ca-bundle.crt \
-        --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
-
-       if [ $? -eq 0 ]; then
-         echo "$RESPONSE"
-         APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
-         echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
-       else
-         echo "$RESPONSE"
-         exit 1
-       fi
+      RESPONSE=$($ETCDCTL_WITH_TLS --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
+      if [ $? -eq 0 ]; then
+        echo "$RESPONSE"
+        APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
+        echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
+      else
+        echo "$RESPONSE"
+        exit 1
+      fi
     }
 
-    start_etcd() {
-      echo "Starting etcd.."
-      mv ${MANIFEST_STOPPED_DIR}/etcd-member.yaml $MANIFEST_DIR
+    # remove member from cluster by name
+    etcd_member_remove() {
+      NAME="$1"
+
+      if [ -z "$NAME" ]; then
+        echo "etcd_member_remove requires 1 argument NAME"
+        exit 1
+      fi
+
+      ID=$($ETCDCTL_WITH_TLS member list | awk -F "," "/\s${NAME}\,/"'{print $1}')
+      if [ "$?" -ne 0 ] || [ -z "$ID" ]; then
+        echo "could not find etcd member $NAME to remove."
+        exit 1
+      fi
+
+      MEMBER_REMOVED=$($ETCDCTL_WITH_TLS member remove $ID -w json | grep "cluster_id")
+      if [ -z "$MEMBER_REMOVED" ]; then
```

Comment:
member remove will exit 0 if the member is not removed, this ended up being the best approach I could find without having access to jq.

Reply:
example return object on success

```
{"header":{"cluster_id":11482253099661102601,"member_id":3676872931145968835,"raft_term":2},"members":[{"ID":12286288530102269298,"name":"etcd-3","peerURLs":["http://etcd-3:2380"],"clientURLs":["http://127.0.0.1:2379","http://etcd-3:2379"]},{"ID":13781855374898664874,"name":"etcd-1","peerURLs":["http://etcd-1:2380"],"clientURLs":["http://127.0.0.1:2379","http://etcd-1:2379"]}]}
```

Reply:
Interesting my testing must of been tainted I can't reproduce. I will update to check return code, thx!